doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,399 @@
+the case of WSL distros on their machine, or VMs running in Azure they may want
+to auto-connect to. These _dynamic_ profiles have a source that might be added
+or removed after the app is installed, and they will be different from user to
+user.
+
+Currently, these profiles are only generated when a user first launches the
+Terminal. If they already have a `profiles.json` file, then we won't run the
+auto-generation behavior. This is obviously not great - if any new types of
+dynamic profiles are added, then users that already have the Terminal installed
+won't get any of these dynamic profiles. Furthemore, if any of the sources of
+these dynamic profiles are removed, then the app won't auto-remove the
+associated profile.
+
+In the new model, with a combined defaults & user settings, how should these
+dynamic profiles work?
+
+I propose we add functionality to automatically search for these profile sources
+and add/remove them on _every_ Terminal launch. To make this functionality work
+appropriately, we'll need to introduce a constraint on dynamic profiles.
+
+**For any dynamic profiles, they must be able to be generated using a stable
+GUID**. For example, any time we try adding the "Ubuntu" profile, we must be
+able to generate the same GUID every time. This way, when a dynamic profile
+generator runs, it can check if that profile source already has a profile
+associated with it, and do nothing (as to not create many duplicate "Ubuntu"
+profiles, for example).
+
+When a dynamic profile generator runs, it will determine what new profiles need
+to be added to the user settings, and it can append those to the list of
+profiles. The generator will return some sort of result indicating that it wants
+a save operation to occur. The app will then save the `profiles.json` file,
+including these new profiles.
+
+If a dynamic profile generator has determined that a profile should no longer be
+used, we don't want to totally remove the profile from the file - perhaps
+there's some data the user would like from those settings. Instead, we'll simply
+mark that profile as `"hidden": true`. That way, the profile will remain in
+their settings, but will not appear in the list of profiles.
+
+Should a dynamic profile generator try to create a profile that's already
+`hidden`, it'll make sure to set `hidden` back to false, making the original
+visible again.
+
+Additionally, a user might not want a dynamic profile generator to always run.
+They might want to keep their Azure connections visible in the list of profiles,
+even if its no longer a valid target. Or they might want to not automatically
+connect to Azure to find new instances every time they launch the terminal. To
+enable scenarios like this, **each dynamic profile generator must provide a way
+to disable it**. For the above listed cases, the two settings might be something
+like `autoloadWslProfiles` and `autoloadAzureConnections`. These will be set to

```

Comment:
future v2 thoughts: if the dynamic provider turns out its pockets and gives us a stable guid, we can just suppress entire provider guids

Reply:
That's an option, though I don't know how we'd expose that clearly to the user...

Reply:
this is a feature that supports a settings UI slightly better than a json file

Reply:
wadda ya know this is something that was added in the #1321 update